tripy/tripy/backend/compiler_api.py

```diff
@@ -33,6 +33,130 @@
 from tripy.utils import json as json_utils
 
 
+# Create default stream.
+@export.public_api(document_under="compiler")
+class default_stream:
+    """
+    Provides access to the default CUDA stream for a given device.
+    This class implements singleton pattern to ensure a single default stream instance per device.
+
+    Args:
+        device (int, optional): The device ID for which to get the default stream. Defaults to the current device.
+
+    Usage:
+        default = default_stream()  # Returns the default Stream object for the current device
+        default_gpu1 = default_stream(1)  # Returns the default Stream object for GPU 1
+
+    Note:
+        Calling default_stream() with the same device always returns the same Stream instance for that device.
+    """
+
+    _instances = {}
+
+    def __new__(cls, device=None):
```

Comment:
Should we use `tp.device` for consistency? We can require it to be a GPU device.

Reply:
Fixed.